packages/patternfly-4/react-charts/src/components/ChartDonut/ChartDonut.js

```diff
@@ -1,21 +1,164 @@
+/* eslint-disable react/require-default-props */
 import React from 'react';
 import PropTypes from 'prop-types';
 import hoistNonReactStatics from 'hoist-non-react-statics';
 import { VictoryPie } from 'victory';
-import { default as ChartTheme } from '../ChartTheme/ChartTheme';
+import ChartLabel from '../ChartLabel/ChartLabel';
+import { styles } from '../ChartTheme/themes/theme-donut';
+import { getDonutTheme } from '../ChartTheme/themes/theme-utils';
 import ChartTooltip from '../ChartTooltip/ChartTooltip';
 
 export const propTypes = {
   /**
    * See TypeScript API docs: https://formidable.com/open-source/victory/docs/victory-area/
    */
-  '': PropTypes.any
+  '': PropTypes.any,
+  /**
+   * The height props specifies the height the svg viewBox of the chart container. This value should be given as a
+   * number of pixels.
+   *
+   * Because Victory renders responsive containers, the width and height props do not determine the width and
+   * height of the chart in number of pixels, but instead define an aspect ratio for the chart. The exact number of
+   * pixels will depend on the size of the container the chart is rendered into.
+   *
+   * Note: innerRadius may need to be set when using this property.
+   */
+  height: PropTypes.number,
+  /**
+   * The subtitle for the donut chart
+   */
+  subTitle: PropTypes.string,
+  /*
+   * Specifies the theme color; blue (default), green, or multi-color. Overridden by the theme property.
+   */
+  themeColor: PropTypes.string,
+  /*
+   * Specifies the theme variant; 'dark' or 'light' (default). Overridden by the theme property.
+   */
+  themeVariant: PropTypes.string,
+  /**
+   * The title for the donut chart
+   */
+  title: PropTypes.string,
+  /**
+   * The width props specifies the width of the svg viewBox of the chart container. This value should be given as a
+   * number of pixels.
+   *
+   * Because Victory renders responsive containers, the width and height props do not determine the width and
+   * height of the chart in number of pixels, but instead define an aspect ratio for the chart. The exact number of
+   * pixels will depend on the size of the container the chart is rendered into.
+   *
+   * Note: innerRadius may need to be set when using this property.
+   */
+  width: PropTypes.number
 };
 
+class ChartDonut extends React.Component {
+  titleRef = React.createRef();
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      height: this.props.height || 0,
+      width: this.props.width || 0
+    };
+  }
+
+  componentDidMount() {
+    // Skip resize handler if height and width were provided
+    if (!(this.state.height || this.state.width)) {
+      setTimeout(() => {
+        window.addEventListener('resize', this.handleResize);
+      });
+      this.handleResize();
+    }
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('resize', this.handleResize);
+  }
+
+  getSubTitle = () => {
```

Comment:
Not sure what the normal pattern is in PF, but could this be a separate `DonutTitle` component instead of a class function?

Reply:
We're already using ChartLabel for this. Perhaps if DonutTitle were exposed via composition, that would be more useful.

Reply:
We could actually get rid of these and the resize listener if we use a VictoryContainer to put the ChartLabel and VictoryPie components into a single svg:

```jsx
<VictoryContainer width={400} height={400}>
    <ChartLabel
      style={[styles.label.title, styles.label.subTitle]}
      text={[title, subTitle]}
      textAnchor="middle"
      verticalAnchor="middle"
      x="200"
      y="200"
    />
     <VictoryPie
      standalone={false}
      labelComponent={<ChartTooltip theme={chartTheme} />}
      theme={chartTheme}
      {...rest}
    />
</VictoryContainer>
```

Reply:
With defaults and calculated props it could look like this:

```jsx
const ChartDonut = (props) => {
  const { height = 200, padAngle = 1, subTitle, themeColor, themeVariant, title, width = 200, ...rest } = props;

  // Define innerRadius and theme separately since default is calculated from other props
  const { innerRadius = (width - 32) / 2, theme = getTheme(themeColor, themeVariant) } = props;

  return (
    <VictoryContainer width={width} height={height}>
      <ChartLabel
        style={[styles.label.title, styles.label.subTitle]}
        text={[title, subTitle]}
        textAnchor="middle"
        verticalAnchor="middle"
        x={width / 2}
        y={height / 2}
      />
      <VictoryPie
        height={height}
        innerRadius={innerRadius}
        labelComponent={<ChartTooltip theme={theme} />}
        padAngle={padAngle}
        standalone={false}
        theme={theme}
        width={width}
        {...rest}
      />
    </VictoryContainer>
  );
};```

Reply:
This is similar to what I've been experimenting with (i.e., ChartContainer). Although, I was having a difficult time without setting innerRadius dynamically. Your calculation seems to work well, thanks.

My only concern with this approach is that it forces the user to provide a height and width. This particular chart would be used differently than all others. I was using the resize listener so the pie chart would behave similarly and continue to be responsive. 

That said, I'll continue to experiment with the listener. It would be ideal if the charts behave the same.

Reply:
Yes, `VictoryContainer` is responsive like the other charts. In my example, I provide default width, height, padAngle, innerRadius,  and theme props so that the user doesn't have to define explicit values for those.